components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -136,6 +143,20 @@ private ClaimUtil() {
                 }
 
                 for (Map.Entry<String, String> entry : userClaims.entrySet()) {
+                    //set local2sp role mappings
+                    if (entry.getKey().equals(FrameworkConstants.LOCAL_ROLE_CLAIM_URI)) {
+                        String domain = IdentityUtil.extractDomainFromName(username);
+                        RealmConfiguration realmConfiguration = realm.getUserStoreManager().getSecondaryUserStoreManager(domain)
+                                .getRealmConfiguration();
+                        String claimSeparator = realmConfiguration
+                                .getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR);
+
+                        String roleClaim = entry.getValue();
+                        List<String> rolesList = new LinkedList<>(Arrays.asList(roleClaim.split(claimSeparator)));
```

Comment:
Because we need a faster remove here. And, we only iterate this List, so random read is not required!